src/app/[locale]/layout.tsx

```diff
@@ -2,13 +2,27 @@ import './globals.css';
 import {useLocale} from 'next-intl';
 import {notFound} from 'next/navigation';
 import {ReactNode} from 'react';
+import Navigation from '@/components/navigation';
+import {useTranslations} from 'next-intl';
+import LocaleSwitcher from '@/components/locale-switcher';
 
 type Props = {
   children: ReactNode;
   params: {locale: string};
 };
 
 export default function RootLayout({children, params}: Props) {
+  const t = useTranslations('Navigation');
+
+  // The navigation is a client component, get the labels first in this server component
+  // See: https://next-intl-docs.vercel.app/docs/next-13/server-components#switching-to-client-components
+  const navigationLabels = {
```

Comment:
Very minor: put this code after the conditions on line 29-31? Then we first check for irregularities (e.g. unknown locale), then set-up the code for handling the regular cases?